### ISSUE TITLE
Implement self-update for non-git Chobby

### ISSUE THREAD

**Issue by gajop** [Spring-Chobby/Chobby#86]
If Chobby has a concrete version, instead of $VERSION, initiate an update if necessary after which the user should be asked to restart the lobby.
Requires Chobby to be put into rapid first https://github.com/spring/RapidTools/issues/43

**Comment by gajop** [comment]
Chobby has been added on rapid, so this should be much easier to do.

**Comment by RafaelLinux** [comment]
Ok, thank you

**Comment by gajop** [comment]
The difficult problem is how we can also upgrade the engine, as rapid only upgrades the mod (Chobby itself), but we also often require a newer engine.
Thoughts (@ashdnazg @abma)? In particular I'm interested about the non-Steam distribution here.

**Comment by RafaelLinux** [comment]
Anyway, if it will simplify this feature, I don't mind to use Steam platform to play thru Chobby, and I think to be in Steam is a good way to promote Zero-K.

**Comment by gajop** [comment]
No. That might be a solution for ZK and/or Steam, but it doesn't work for Spring.

**Comment by abma** [comment]
> The difficult problem is how we can also upgrade the engine, as rapid only upgrades the mod (Chobby itself), but we also often require a newer engine.

1. connect to lobby server to get the current stable engine version (TASServer ...)
2. download new engine version via prdownloader api: "spring stable version" should work
3. start the downloaded version

only 3 is missing afaik.

**Comment by abma** [comment]
basicly this report is a duplicate of #53

**Comment by gajop** [comment]
I'd say it's just related (or depends) on it. You still need to create a rapid self-update as well as the entire GUI.
I don't think we have a very good API to figure out if there is a newer version available. 
Also, currently we want to store the engine version required for Chobby in modinfo.lua (https://github.com/Spring-Chobby/Chobby/blob/master/modinfo.lua#L21), and in that case we'd need to read that value from the newly downloaded Chobby.
